refactor: remove unneeded .stackblitzrc, update template dependencies

fvsch opened the pull request:
This PR harmonizes what basic config files we find in a starter template.

### What's in `package.json`

Almost all starters have a `package.json` file. I've tried to harmonize those files to follow this pattern:

```json
{
  "name": "{folder-name}-demo",
  "version": "0.0.0",
  "private": true,
  "scripts": { … },
  "dependencies": { … },
  "devDependencies": { … }
}
```

Some templates may not have `"scripts"` (most for some EngineBlock starters), some may not have `"devDependencies"`.

Open questions:

1. What should we use for the package name?
    - Should we include one at all?
    - Is the `"{folder-name}-demo"` pattern okay, or do we want to use “starter” or “template” instead of “demo” maybe?
    - Note that if we end up using a name that matches an existing package on npm, running `npm info` will show information for that package merged with info from the local `package.json`, which is quite strange, but maybe a very rare problem.
2. Is the `"version"` even useful? I’m tempted to just drop it.

### Removing all the `.stackblitzrc` files

All the `.stackblitzrc` files in WebContainer projects were spelling out defaults, such as `{ "installDependencies": true, startCommand: "npm run dev" }`.

I removed them for the sake of readability, but if we want to include one with defaults in every WC starter template we could do that too.

### Making sure every WC template has a `.gitignore`

For every WebContainer-based template, I made sure it had a `.gitignore` file that listed at least:

- `node_modules`
- whatever caching folder the framework is using (`.next`, `.nuxt`, `.svelte-kit`, etc.)
- whatever build output folder resulted from running `npm run build` (`build`, `dist`, etc.)

### Harmonizing `index.html` files

- Made sure every one had a `<!DOCTYPE html>`
- Removed `<meta http-equiv="X-UA-Compatible" content="IE=edge">`
- Used the same minimal viewport meta everywhere: `<meta name="viewport" content="width=device-width">`

### Harmonizing base CSS styles

We had base `style.css` files which looked like:

```css
h1, h2 {
  font-family: Lato;
}
```

That's not super useful, especially since the preview documents don't load the Lato fonts at all. (They used to, in EngineBlock projects, but we removed those fonts in early 2022 for performance reasons.)

Instead I'm proposing a very basic stylesheet looking like this:

```css
* {
  box-sizing: border-box;
}

body {
  margin: 0;
  padding: 1rem;
  font-family: system-ui, sans-serif;
  color: black;
  background-color: white;
}

h1 {
  font-weight: 800;
  font-size: 1.5rem;
}
```

fvsch commented on bootstrap-5/package.json:
```diff
@@ -1,19 +1,17 @@
   "version": "0.0.0",
+  "private": true,
   "scripts": {
-    "start": "webpack serve"
+    "dev": "vite",
+    "build": "vite build",
+    "preview": "vite preview"
   },
```
Converted the Bootstrap 5 example from webpack to Vite.
Simpler less verbose config, and HMR for free.

fvsch commented on express-simple/README.md:
```diff
@@ -1,3 +1,7 @@
+
+This project shows a simple express server serving a single HTML page and using `express.static` to serve static files.
+
+Check out the [express documentation](https://expressjs.com/) for more information.
```
Apparently we had just saved an automatically generated `README.md` pushed to GitHub by the StackBlitz editor.

fvsch commented on gsap-react/package.json:
```diff
@@ -1,19 +1,22 @@
-  }
+    "react-scripts": "^5.0.1"
+  },
+  "browserslist": [
+    "defaults"
+  ]
```
The `browserslist` config is required by `react-scripts`.

fvsch commented on js/tsconfig.json:
```diff
@@ -1,5 +0,0 @@
-{
-  "compilerOptions": {
-    "target": "esnext"
-  }
-}
```
I’m not sure why the JS EngineBlock starter template would need a `tsconfig.json`. My understanding is that is doesn't.

fvsch commented on node/package.json:
```diff
@@ -1,7 +1,8 @@
   "version": "0.0.0",
+  "private": true,
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "start": "node index.js"
```
Instead of telling users to run `node index.js` in the Terminal as a comment in `index.js`, we can just run it on startup.

sulco reviewed (approved):
I like all the changes very much! I'd drop `version` as well, but I don't have strong feelings about that. 

For the package name, I think the `-starter` makes a bit more sense, keeping it in the same "name space" as we're already using when calling this pieces of StackBlitz experience a "starter".

fvsch commented:
> I'd drop `version` as well, but I don't have strong feelings about that.

Dropped it. I agree that it's not useful for projects.

> For the package name, I think the `-starter` makes a bit more sense

Changed from `-demo` suffix to `-starter` suffix.
